Modules/@babylonjs/react-native-basekit/scripts/install.js

```diff
@@ -0,0 +1,174 @@
+
+  return [site, package.version, binaryFilename].join('/');
+}
+
+const downloadExtractAndCache = (url, cachedFilePath) => {
+  const options = {
+    followRedirects: true, // Follow HTTP 3xx redirects
+  };
+
+  https.get(url, options, (response) => {
+    if ([301, 302, 303, 307, 308].includes(response.statusCode)) {
+      // If the response is a redirect, recursively call downloadExtractAndCache with the new URL
+      downloadExtractAndCache(response.headers.location, cachedFilePath);
+    } else if (response.statusCode === 200) {
+      const gunzip = zlib.createGunzip();
+      const untar = tar.extract({ cwd: process.cwd() }); // Extract to the current working directory
+
+      // create cache directory to store download file
+      try {
+        fs.mkdirSync(path.dirname(cachedFilePath), {recursive: true});
+      } catch (err) {
+        console.error('Unable to create directory for cache', err);
+        process.exit(1);
+      }
+
+      const chunks = [];
+      response.on('data', (chunk) =>{
+        chunks.push(chunk);
+      })
+      .on('end', (err) => {
+        const fileData = Buffer.concat(chunks);
+        fs.writeFile(cachedFilePath, fileData, (err) => {
+          if (err) {
+            console.error('Error writing BabylonReactNative cache file:', err);
+            process.exit(1);
+          }
+        });
+      })
+      .pipe(gunzip) // Unzip the response
+      .pipe(untar)  // Extract the tar file
+      .on('error', (err) => {
+        console.error('Error extracting the tar.gz file:', err);
+        process.exit(1);
+      });
+    } else {
+      console.error('Failed to download the file. Status code:', response.statusCode);
+      process.exit(1);
+    }
+  })
+  .on('error', (err) => {
+    console.error('Error downloading the file:', err);
+    process.exit(1);
+  });
+};
+
+function Install() {
+  // get Babylon React Native package version
+  const packageJsonPath = process.env.npm_package_json;
+  const packageJson = require(packageJsonPath);
+  if (!packageJson.version) {
+    console.err("Babylon React Native version not found");
+    process.exit(1);
+  }
+  // get user project react-native version
+  const projectPackageJsonPath = process.env.npm_config_local_prefix + '/package.json';
+
+  // Read the project's package.json file
+  const projectPackageJson = require(projectPackageJsonPath);
+  // Check if the project's package.json has a 'react-native' dependency
+  if (projectPackageJson.dependencies && projectPackageJson.dependencies['react-native']) {
+    const reactNativeVersion = projectPackageJson.dependencies['react-native'];
+    console.log(`Detected React Native version: ${reactNativeVersion}`);
+    reactNativePostfix = "";
+    if (reactNativeVersion >= '0.71') {
+      reactNativePostfix = "0.71";
+    } else if (reactNativeVersion >= '0.70') {
+      reactNativePostfix = "0.70";
+    } else if (reactNativeVersion >= '0.69') {
+      reactNativePostfix = "0.69";
```

Comment:
do you mean to compare strings here? I mean, it's javascript and it will (probably as expected) work because of coercion,  but using numbers directly would make more sense

Reply:
reactNativeVersion can be '0.71.X', would it coerce to 0.71 ?

Reply:
hm.  That's incorrect comparison. It is true that `"0.7.X > "0.7"` is true, but this is false: `"0.10.X > "0.7"`. It works well until 9 because of the way string comparison works in JS.